### DISCUSSION TITLE
SOF VirtIO: Buliding Linux with sof-host-defconfig & sof-guest-defconfig

### DISCUSSION THREAD

**Discussion by Groundation** [thesofproject/sof#8390] [General]
Hello, 

SOF noobie here trying to run SOF emulation on a virtual machine Ubuntu 22.04. 

I've been following these instructions to make Linux VM image: https://thesofproject.github.io/latest/developer_guides/virtualization/running.html 

It says _"Use the sof-host-defconfig and sof-guest-defconfig [kernel configuration](https://github.com/thesofproject/kconfig) patches to configure the respective kernels."_, but I am not able to find any of these patches. 

Am I reading this wrong and can anybody help me with configuring Linux kernel for host/guest build?

Nenad

**Comment by lgirdwood** [comment]
@Groundation sorry, not sure where those patches are, but this information is quite old now (and should be updated). Virtual audio has now been upstreamed into ALSA. i.e. you should be able to use Qemu and virtual ALSA audio with any host PC sound device (limited to HDA like uses cases).

**Reply by Groundation** [reply]
Hi Liam, thank you for the answer. 

I have been following official SOF documentation and have been trying to make a SOF emulation on my PC with the information available in the docs (which is btw hard to follow without deeper knowledge of the system). Is there any more documentation or explanation to follow for us noobies? 

For example, I have been having difficult times trying to build QEMU from SOF branch, and my guess the problem is with GCC version (having some array subscript build errors) since I am using GCC 11 while I probably should be using some older GCC version, maybe 7 or similar. But, I am not able to find those type of information in the official doc.   

Either way, I would be thankful if you could direct me to some up to date documentation and general description on how I could run the SOF complete system on a virtual machine just as a way of learning the mechanism and possibly running it on a specific hardware at some point later. 

Help would be much appreciated 
Nenad

**Reply by lgirdwood** [reply]
The xtensa qemu code used by SOF is quite old now (IIRC qemu 5.x), it was used for CI purposes (basic smoke tests for PRs) and have been deprecated in CI as we have better/faster methods today. The code has bitrotted a lot as Qemu has moved quickly and so has virtio ALSA.

To the best of my knowledge (and I've not checked this for about 2 years), you will probbaly need the latest Qemu and Pipewire/Pulseaudio (if they have virtio audio support). The soundserver then would take care of sending the virtio audio to/from Qemu to the ALSA audio card.
